dhcpv4/server4/conn.go

```diff
@@ -14,7 +14,7 @@ import (
 // given based on a IPv4 DGRAM socket. The UDP connection allows broadcasting.
 //
 // The interface must already be configured.
-func NewIPv4UDPConn(iface string, port int) (*net.UDPConn, error) {
+func NewIPv4UDPConn(iface string, addr *net.UDPAddr) (*net.UDPConn, error) {
```

Comment:
Make addr net.UDPAddr so it can't be nil, or check it for nil?

Reply:
Good point. NewServer() also doesn't check for nil and takes a pointer, I'll add in both

Reply:
In the end I didn't add a check in NewServer, since it now just passes the value through it won't crash with addr=nil anymore (but would have before this PR). NewIPv4UDPConn now checks and will do the right thing